kotlinx-coroutines-core/common/test/flow/channels/ChannelBuildersFlowTest.kt

```diff
@@ -9,6 +9,51 @@ import kotlinx.coroutines.channels.*
 import kotlin.test.*
 
 class ChannelBuildersFlowTest : TestBase() {
+    @Test
+    fun testChannelConsumeAsFlow() = runTest {
+        val channel = produce {
+           repeat(10) {
+               send(it + 1)
+           }
+        }
+        val flow = channel.consumeAsFlow()
+        assertEquals(55, flow.sum())
+        assertFailsWith<IllegalStateException> { flow.collect() }
+    }
+
+    @Test
+    fun testConsumeAsFlowCancellation() = runTest {
+        expect(1)
+        val channel = produce(NonCancellable) { // otherwise failure will cancel scope as well
+            repeat(10) {
+                send(it + 1)
+            }
+            throw TestException()
+        }
+        val flow = channel.consumeAsFlow()
+        assertEquals(15, flow.take(5).sum())
+        // the channel should have been canceled, even though took only 5 elements
+        assertTrue(channel.isClosedForReceive)
+        assertFailsWith<IllegalStateException> { flow.collect() }
+        finish(2)
+    }
+
+    @Test
+    fun testConsumeAsFlowException() = runTest {
+        expect(1)
+        val channel = produce(NonCancellable) { // otherwise failure will cancel scope as well
+            repeat(10) {
+                send(it + 1)
+            }
+            throw TestException()
+        }
+        val flow = channel.consumeAsFlow()
+        assertFailsWith<TestException> { flow.sum() }
+        assertFailsWith<IllegalStateException> { flow.collect() }
+        finish(2)
```

Comment:
not sure why these `expect(1)` + `finish(2)` are needed

Reply:
Not really needed. Just for consistency.